src/runtime_src/core/common/device.cpp

```diff
@@ -78,6 +78,53 @@ register_axlf(const axlf* top)
 
   // Build modified CONNECTIVITY and MEM_TOPOLOGY section based on memory group ids
   // Base groups off data from driver
+  auto m_itr = m_axlf_sections.find(MEM_TOPOLOGY);
+  int g_count = (int)m_grp_info.size();
+  if (g_count && (m_itr != m_axlf_sections.end())) {
+    auto m_mem = reinterpret_cast<::mem_topology*>((*m_itr).second.data());
```

Comment:
Can you use get_axlf_section<const mem_topology*>() ?

Reply:
The return type of get_axlf_section if std::pair<**const char***, size_t>. But in our case, we need to modify the mem_topology, which is not allowed.

Reply:
I see.  This makes sense.  I still think you should use the existing method and maybe const_cast the data, or alternatively provide a non-const private method that does just that.